src/ripple/shamap/SHAMap.h

```diff
@@ -358,66 +362,109 @@ SHAMap::setUnbacked ()
 
 //------------------------------------------------------------------------------
 
-class SHAMap::iterator
-    : public boost::iterator_facade<
-        SHAMap::iterator,
-            std::shared_ptr<SHAMapItem const> const,
-                std::forward_iterator_tag>
+class SHAMap::const_iterator
 {
-private:
-    friend class boost::iterator_core_access;
+public:
+    using iterator_category = std::forward_iterator_tag;
+    using difference_type   = std::ptrdiff_t;
+    using value_type        = SHAMapItem;
+    using reference         = value_type const&;
+    using pointer           = value_type const*;
 
-    SHAMap const* map_ = nullptr;
-    std::shared_ptr<
-        SHAMapItem const> item_;
+private:
+    NodeStack     stack_;
+    SHAMap const* map_  = nullptr;
+    pointer       item_ = nullptr;
 
 public:
-    iterator() = default;
-    iterator (iterator const&) = default;
-    iterator& operator= (iterator const&) = default;
+    const_iterator() = default;
```

Comment:
Thanks.  Right.  The private constructors would disable the default constructor.  I missed that.  Thanks for the explanation.